test/integration/concierge_tls_spec_test.go

```diff
@@ -0,0 +1,221 @@
+			expectedError:      `The WebhookAuthenticator "invalid-webhook-auth-invalid-kind" is invalid: spec.tls.certificateAuthorityDataSource.kind: Unsupported value: "": supported values: "Secret", "ConfigMap"`,
+		},
+		{
+			name: "should disallow certificate authority data source with invalid kind",
+			customResourceYaml: []byte(here.Doc(`
+			---
+			apiVersion: authentication.concierge.pinniped.dev/v1alpha1
+			kind: WebhookAuthenticator
+			metadata:
+				name: invalid-webhook-auth-invalid-kind
+			spec:
+				endpoint: "https://web-auth/token"
+				tls:
+					certificateAuthorityDataSource:
+						kind: sorcery
+						name: foo
+						key: bar
+			`)),
+			customResourceName: "invalid-webhook-auth-invalid-kind",
+			expectedError:      `The WebhookAuthenticator "invalid-webhook-auth-invalid-kind" is invalid: spec.tls.certificateAuthorityDataSource.kind: Unsupported value: "sorcery": supported values: "Secret", "ConfigMap"`,
+		},
+		{
+			name: "should create a custom resource passing all validations using a Secret source",
+			customResourceYaml: []byte(here.Doc(`
+			---
+			apiVersion: authentication.concierge.pinniped.dev/v1alpha1
+			kind: WebhookAuthenticator
+			metadata:
+				name: valid-webhook-auth-secret-kind1
+			spec:
+				endpoint: "https://web-auth/token"
+				tls:
+					certificateAuthorityDataSource:
+						kind: Secret
+						name: foo
+						key: bar
+		`)),
+			customResourceName: "valid-webhook-auth-secret-kind1",
+			expectedError:      "",
+		},
+		{
+			name: "should create a custom resource passing all validations using a ConfigMap source",
+			customResourceYaml: []byte(here.Doc(`
+			---
+			apiVersion: authentication.concierge.pinniped.dev/v1alpha1
+			kind: WebhookAuthenticator
+			metadata:
+				name: valid-webhook-auth-cm-kind1
+			spec:
+				endpoint: "https://web-auth/token"
+				tls:
+					certificateAuthorityDataSource:
+						kind: ConfigMap
+						name: foo
+						key: bar
+			`)),
+			customResourceName: "valid-webhook-auth-cm-kind1",
+			expectedError:      "",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			resourceName := tc.customResourceName
+			expectedError := tc.expectedError
+			yamlBytes := tc.customResourceYaml
+			t.Parallel()
+			yamlFilepath := filepath.Join(t.TempDir(), fmt.Sprintf("tls-spec-validation-%s.yaml", resourceName))
+
+			require.NoError(t, os.WriteFile(yamlFilepath, yamlBytes, 0600))
+			cmd := exec.CommandContext(context.Background(), "kubectl", []string{"apply", "-f", yamlFilepath}...)
+			var stdOut, stdErr bytes.Buffer
+			cmd.Stdout = &stdOut
+			cmd.Stderr = &stdErr
+
+			err := cmd.Run()
+			if expectedError == "" {
+				require.NoError(t, err)
+				require.Equal(t, fmt.Sprintf("webhookauthenticator.authentication.concierge.pinniped.dev/%s created\n", resourceName), stdOut.String())
+				require.Empty(t, stdErr.String())
+				require.NoError(t, exec.Command("kubectl", []string{"delete", "-f", yamlFilepath}...).Run())
```

Comment:
Cleanups should happen using `t.Cleanup` to ensure that they will be attempted even when a test fails.